constructor-manager-ui/src/constructor_manager_ui/main.py

```diff
@@ -1,458 +1,92 @@
-    ):
-        super().__init__(parent=parent)
-        self.package_name = package_name
-        self.current_version = install_information["current_version"]
-        self.snapshot_version = install_information["snapshot_version"]
-        self.updates_widget = None
-        self.packages_tablewidget = None
-        self.setWindowTitle(f"{package_name} installation manager")
-        self.setMinimumSize(QSize(500, 500))
-        self.setup_layout()
+logger = logging.getLogger(__name__)
 
-    def _create_install_information_group(self):
-        install_information_group = QGroupBox("Install information")
-        install_information_layout = QVBoxLayout()
-        current_version_layout = QHBoxLayout()
 
-        # Current version labels and button
-        current_version_label = QLabel(
-            f"{self.package_name} {self.current_version['version']}"
-        )
-        last_modified_version_label = QLabel(
-            f"Last modified {self.current_version['last_modified']}"
-        )
-        current_version_open_button = QPushButton("Open")
-        current_version_open_button.setObjectName("open_button")
-        current_version_layout.addWidget(current_version_label)
-        current_version_layout.addSpacing(10)
-        current_version_layout.addWidget(last_modified_version_label)
-        current_version_layout.addSpacing(10)
-        current_version_layout.addWidget(current_version_open_button)
-        current_version_layout.addStretch(1)
-        install_information_layout.addLayout(current_version_layout)
+def dedup(items: Tuple[Any, ...]) -> Tuple[Any, ...]:
+    """Deduplicate an list of items."""
+    new_items: Tuple[Any, ...] = ()
+    for item in items:
+        if item not in new_items:
+            new_items += (item,)
 
-        # Line to divide current section and update section
-        install_version_line = QFrame()
-        install_version_line.setObjectName("separator")
-        install_version_line.setFrameShape(QFrame.HLine)
-        install_version_line.setFrameShadow(QFrame.Sunken)
-        install_version_line.setLineWidth(1)
-        install_information_layout.addWidget(install_version_line)
+    return new_items
 
-        # Update information widget
-        self.updates_widget = UpdateWidget(self.package_name, parent=self)
-        install_information_layout.addWidget(self.updates_widget)
-        install_information_group.setLayout(install_information_layout)
 
-        # Signals
-        # Open button signal
-        current_version_open_button.clicked.connect(self.open_installed)
-        # Update widget signals
-        self.updates_widget.install_version.connect(self.install_version)
-        self.updates_widget.skip_version.connect(self.skip_version)
+def _configure_logging(log_level="WARNING"):
+    """Configure logging."""
+    import constructor_manager_api  # type: ignore
 
-        return install_information_group
+    log_level = getattr(logging, log_level.upper())
```

Comment:
Tried running `constructor-manager-ui napari --current-version 0.4.16 --build-string pyside --plugins-url https://api.napari-hub.org/plugins --channel conda-forge` locally and got the following traceback related to this line:

```python-traceback
Traceback (most recent call last):
  File "C:\Users\dalth\anaconda3\envs\constructor-manager\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\dalth\anaconda3\envs\constructor-manager\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\dalth\anaconda3\envs\constructor-manager\Scripts\constructor-manager-ui.exe\__main__.py", line 7, in <module>
  File "E:\Acer\Documentos\Quansight\Napari\packaging\constructor-manager-ui\src\constructor_manager_ui\main.py", line 58, in run
    _configure_logging(settings["log"])
  File "E:\Acer\Documentos\Quansight\Napari\packaging\constructor-manager-ui\src\constructor_manager_ui\main.py", line 33, in _configure_logging
    log_level = getattr(logging, log_level.upper())
AttributeError: 'NoneType' object has no attribute 'upper'
```

Reply:
This is part of the issue of calling on windows. Fixing over the CLI PR